src/app/core/default-pattern-renderer/default-pattern-renderer.component.html

```diff
@@ -1,13 +1,22 @@
 <mat-progress-spinner *ngIf="isLoading" mode="indeterminate"></mat-progress-spinner>
 <mat-card class="pattern-card mat-elevation-z5">
   <mat-card-header (mouseenter)="showActionButtons = true" (mouseleave)="showActionButtons = false">
-    <pp-action-button-bar [displayText]="pattern?.name" [firstAddButton]="false" [iconEdit]="true"
+    <pp-action-button-bar [displayText]="pattern?.name" [firstAddButton]="false"
+                          [iconEdit]="true"
                           [iconUrl]="pattern?.iconUrl" (iconEditClicked)="editIcon()">
     </pp-action-button-bar>
     <span> <button mat-stroked-button style="margin-left: 5px" color="primary" *ngIf="showActionButtons"
                    (click)="editIcon()" matTooltip="Edit"><i class="material-icons">mode_edit</i></button>
   </span>
   </mat-card-header>
+
+  <span class="copyright" *ngIf="pattern?.paperRef"> <i class="material-icons">copyright</i>
+    <!--    Display either a textual reference for yet unpublished papers or a html link -->
+    {{pattern?.paperRef.startsWith('http') ?
+      "The original publication of this pattern can be found&nbsp;" :
```

Comment:
Maybe also add "in" here?

Reply:
so the reference would be "...can be found in [here](http://quantumcomputingpatterns.com)"?

Reply:
Ah, you have the if twice. You decide :)